ee/tabby-schema/src/schema/mod.rs

```diff
@@ -658,6 +673,50 @@ impl Query {
 
         Ok(SourceIdAccessPolicy { source_id, read })
     }
+
+    async fn test_model_connection(ctx: &Context, backend: ModelHealthBackend) -> Result<i32> {
```

Comment:
would it be better to create a new error type for Result (Result<i32, TestModelConnectionError>)? e.g we might wanna report different error type to user in future (e.g credential error, timeout, connection failed, ...)

Reply:
All the GraphQL APIs return the CoreError, so I added a field `ModelConnectionFailed(String)` in CoreError

- return NotFound, when the model backend is not running
- return ModelConnectionFailed with error message, when failed to connect model